api/client/import.go

```diff
@@ -16,30 +18,68 @@ import (
 //
 // The URL argument is the address of a tarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz) file. If the URL is '-', then the tar file is read from STDIN.
 //
-// Usage: docker import [OPTIONS] URL [REPOSITORY[:TAG]]
+// Usage: docker import [OPTIONS] [-i URL] [REPOSITORY[:TAG]]
 func (cli *DockerCli) CmdImport(args ...string) error {
-	cmd := cli.Subcmd("import", "URL|- [REPOSITORY[:TAG]]", "Create an empty filesystem image and import the contents of the\ntarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz) into it, then\noptionally tag it.", true)
+	cmd := cli.Subcmd("import", "[REPOSITORY[:TAG]]", "Create an empty filesystem image and import the contents of the\ntarball (.tar, .tar.gz, .tgz, .bzip, .tar.xz, .txz) into it, then\noptionally tag it.", true)
+	infile := cmd.String([]string{"i", "-input"}, "", "Read from a tar archive file, instead of STDIN")
 	flChanges := opts.NewListOpts(nil)
 	cmd.Var(&flChanges, []string{"c", "-change"}, "Apply Dockerfile instruction to the created image")
-	cmd.Require(flag.Min, 1)
+	cmd.Require(flag.Min, 0)
 
 	utils.ParseFlags(cmd, args, true)
 
 	var (
-		v          = url.Values{}
-		src        = cmd.Arg(0)
-		repository = cmd.Arg(1)
+		input      io.Reader = cli.in
+		err        error
+		src        string
+		repository string = cmd.Arg(1)
+		v                 = url.Values{}
 	)
+	if *infile != "" {
+
+		repository = cmd.Arg(0)
+
+		if strings.HasPrefix(*infile, "http://") || strings.HasPrefix(*infile, "https://") {
+			src = *infile
+			input = nil
+		} else if strings.HasPrefix(*infile, "file://") {
```

Comment:
I should add, I'm not fond of the file:// format. While its possible to reference local files that way, I don't really think its natural for people to do so when typing on the cmd line.  If we were in a situation where we had a whole like of URLs to process, or URLs were the main way to represent a resource, then I think it might make sense from a consistency perspective, but I doubt most people would naturally prefer to type `file://foo.tar` over just `foo.tar`